shared/data-scripts/user-data-server.sh

```diff
@@ -15,31 +15,73 @@ sed -i "s/CONSUL_TOKEN/${nomad_consul_token_secret}/g" /etc/nomad.d/nomad.hcl
 sudo systemctl restart nomad
 
 echo "Finished server setup"
-sleep 10
-echo "ACL bootstrap begin"
-
-# Bootstrap consul ACLs
-consul acl bootstrap | grep -i secretid | awk '{print $2}' > $CONSUL_BOOTSTRAP_TOKEN
-
-if [ $? -eq 0 ]; then
-    consul acl policy create -name 'nomad-auto-join' -rules="@$ACL_DIRECTORY/consul-acl-nomad-auto-join.hcl" -token-file=$CONSUL_BOOTSTRAP_TOKEN
-
-    consul acl role create -name "nomad-auto-join" -description "Role with policies necessary for nomad servers and clients to auto-join via Consul." -policy-name "nomad-auto-join" -token-file=$CONSUL_BOOTSTRAP_TOKEN
-
-    consul acl token create -accessor=${nomad_consul_token_id} -secret=${nomad_consul_token_secret} -description "Nomad server/client auto-join token" -role-name nomad-auto-join -token-file=$CONSUL_BOOTSTRAP_TOKEN
 
-    # Wait for nomad servers to come up
-    sleep 40
-
-    # Bootstrap nomad ACLs
-    nomad acl bootstrap | grep -i secret | awk -F '=' '{print $2}' | xargs > $NOMAD_BOOTSTRAP_TOKEN
-
-    nomad acl policy apply -token $(cat $NOMAD_BOOTSTRAP_TOKEN) -description "Policy to allow reading of agents and nodes and listing and submitting jobs in all namespaces." node-read-job-submit $ACL_DIRECTORY/nomad-acl-user.hcl
-
-    nomad acl token create -token $(cat $NOMAD_BOOTSTRAP_TOKEN) -name "read-token" -policy node-read-job-submit | grep -i secret | awk -F "=" '{print $2}' | xargs > $NOMAD_USER_TOKEN
+echo "ACL bootstrap begin"
 
-    # Write user token to kv
-    consul kv put -token-file=$CONSUL_BOOTSTRAP_TOKEN nomad_user_token $(cat $NOMAD_USER_TOKEN)
-fi
+# Wait until leader has been elected and bootstrap consul ACLs
+while true; do
+    # capture stdout and stderr
+    set +e
```

Comment:
Nothing major but you might want to invert `set -e` and `set +e` if you want to debug the bootstrap part and not the outer part of it.

Reply:
My bad, confused between `-s` and `-x`. Ignore this comment